Authentication/Authentication/DuplicateUserCredentialsException.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Authentication
+{
+    class DuplicateUserCredentialsException : Exception
+    {
+        public DuplicateUserCredentialsException()
+        {
+            Console.WriteLine("There are duplicate entries in the file");
```

Comment:
Don't do this- in some environments we won't be able to see the exception message.
Instead, use this: ` public DuplicateUserCredentialsException(): base("There are duplicate entries in the file"){}`

Reply:
In fact, specify the duplicate entry.